File: java/com/example/spottio/GroupAdapter.java
package com.example.spottio;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import java.util.List;

public class GroupAdapter extends ArrayAdapter<GroupInfo> {

    public GroupAdapter(Context context, List<GroupInfo> groups) {
        super(context, 0, groups);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(android.R.layout.simple_list_item_1, parent, false);
        }

        GroupInfo group = getItem(position);
        TextView text = convertView.findViewById(android.R.id.text1);

        if (group != null && text != null) {
            text.setText(group.getName());
            text.setTextSize(18);
            text.setPadding(24, 32, 24, 32);
        }

        return convertView;
    }
}

File: java/com/example/spottio/GroupAdapter.kt
package com.example.spottio

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

class GroupAdapter(context: Context, groups: List<GroupInfo>) : ArrayAdapter<GroupInfo>(context, 0, groups) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        // L'operatore Elvis (?:) dice: usa convertView, se è nullo inflata la vista
        val view = convertView ?: LayoutInflater.from(context).inflate(android.R.layout.simple_list_item_1, parent, false)

        val group = getItem(position)
        val text = view.findViewById<TextView>(android.R.id.text1)

        if (group != null && text != null) {
            text.text = group.name
            text.textSize = 18f
            text.setPadding(24, 32, 24, 32)
        }

        return view
    }
}